archive/archive_ingest/src/archive_ingest.py

```diff
@@ -0,0 +1,68 @@
+# -*- encoding: utf-8 -*-
+"""
+Receives a message to ingest a bag giving the URL and publishes the archive event to an SNS topic.
+"""
+
+import os
+
+import boto3
+import daiquiri
+from urllib.parse import urlparse
+
+from wellcome_aws_utils.sns_utils import publish_sns_message
+from wellcome_aws_utils.lambda_utils import log_on_error
+
+daiquiri.setup(level=os.environ.get('LOG_LEVEL', 'INFO'))
+logger = daiquiri.getLogger()
+
+
+def ensure_valid_request(event):
+    required_key = 'bagURL'
+    if type(event) is not dict:
+        raise ValueError(f"Invalid request not json dict: {event}")
+    if required_key not in event:
+        raise ValueError(f"Invalid request missing {required_key}")
+
+
+def archive_bag_message(bag_url):
+    """
+    Generates bag archive messages.
+    """
+    url = urlparse(bag_url)
+    if url.scheme == 's3':
+        bucket = url.netloc
+        key = url.path.lstrip('/')
+        return {
+            'namespace': bucket,
+            'key': key
+        }
+    else:
+        raise ValueError(f"Unrecognised url scheme: {bag_url}")
+
+
+@log_on_error
+def handler(event, _ctx=None, sns_client=None):
+    logger.info(f"received {event}")
```

Comment:
We don't log the received event in our other Lambdas (except perhaps at debug) unless there's an error – that’s what the `@log_on_error` does for us. This keeps logging costs down.

Also, good practice with `logger` is to use `%`-formatters, and pass them as extra arguments to the log method, i.e.:

```python
logger.info('received %r', event)
```

If the log message isn't used (because there are no handlers, or the level is too low), you save the cost of serialising the object as a string. Probably not a big issue here, but a good habit to get into.